packages/ra-ui-materialui/src/input/DateTimeInput.tsx

```diff
@@ -15,7 +16,7 @@ const leftPad2 = leftPad(2);
  * @returns {String} A standardized datetime (yyyy-MM-ddThh:mm), to be passed to an <input type="datetime-local" />
  */
 const convertDateToString = v => {
-    if (!(v instanceof Date) || isNaN(v)) return '';
+    if (!(v instanceof Date) || isNaN(v.getDate())) return '';
```

Comment:
I'm confused about `v.getDate` here..

If `v` is a Date, this function returns the day of the month, which is always a number
If `v` is a date from moment or date-fns, it returns a Date, and is never a number

Am I missing something?

Also, the `convertDateToString` doesn't respect the ISO 8601 standard: it misses the timezone at the end (should be `Z` or `+00` something).

All thing considered, can we replacethis whole function by `date.toISOString()`, or do we need to strip the timezone?

Reply:
I haven't think about it, just trying to convert those inputs to TypeScript and make them use the hook from react-final-form, without changing too many things.

Reply:
If you think this is a prb, open an issue and we'll address it in a dedicated PR